### PR TITLE
Enable overriding and extending list value

### PR DESCRIPTION
**Pull request by JordanP**
With this commit, it's now possible to set a value to a specific item in
a list with the CLI: "kuku render ./templates --set a.0=foo"

Code is a bit tricky, prepare a pen if you want to proof-read it.
Otherwise, reading the unit tests should give a good confidence level.

### CODE REVIEW

File: kuku/tests/test_values.py [hunk 1]
```diff
@@ -39,13 +39,29 @@ def test_nested_dicts(values, expected):
     "values, expected",
     [
         (["a.0.b=v1", "a.0.c=v2"], {"a": [{"c": "v2"}]}),
-        # (["a.0.b=v1", "a.1.c=v2"], [{"a": [{"c": "v2"}]}, {"a": [{"b": "v1"}]}]),
+        (["a.0.b=v1", "a.1.c=v2"], {"a": [{"b": "v1"}, {"c": "v2"}]}),
```
Comment (xarg) [inline kuku/tests/test_values.py:43]: let's add a test for replacement here as well. Optionally we'd have an "invalid" index error test as well.

Comment (JordanP) [reply kuku/tests/test_values.py:43]: I added both (a couple of lines below)

Comment (xarg) [reply kuku/tests/test_values.py:43]: I saw them, but those are for file replacement, not values.

Comment (JordanP) [reply kuku/tests/test_values.py:43]: You are right. I added 2 new test cases.

**Pull request merged**